keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/Resource.kt

```diff
@@ -23,21 +22,25 @@ data class Resource<T : ResourceSpec>(
 
     other as Resource<*>
 
-    if (apiVersion != other.apiVersion) return false
     if (kind != other.kind) return false
     if (spec != other.spec) return false
 
     return true
   }
 
   override fun hashCode(): Int {
-    var result = apiVersion.hashCode()
-    result = 31 * result + kind.hashCode()
+    var result = kind.hashCode()
     result = 31 * result + spec.hashCode()
     return result
   }
 }
 
+/**
+ * Creates a resource id in the correct format.
+ */
+fun assignId(kind: String, spec: ResourceSpec) =
+  "${kind.substringBefore("/")}:${kind.substringAfterLast("/")}:${spec.id}"
+
```

Comment:
Created a function to standardize id format as ids with `/` would break URLs.

Reply:
Shouldn't we just do this in the constructor instead?

Reply:
Also, won't colons be confusing with resource IDs since those use colons as well? Maybe `my.api@v1:my:resource` or just `my.api.v1:my:resource`?

Reply:
this is resource ids